### PR TITLE
Allow force restore for Transit Key Restores

### PR DESCRIPTION
**Pull request by catsby**
Adds an optional parameter `force` to the `Restore` operation of Transit keys, allowing a user to force the restoration even a key by the same name already exists. The default is false, and the current behavior remains unchanged. 

Implements https://github.com/hashicorp/vault/issues/5156

### CODE REVIEW

File: helper/keysutil/lock_manager.go [hunk 1]
```diff
@@ -103,26 +103,50 @@ func (lm *LockManager) RestorePolicy(ctx context.Context, storage logical.Storag
...
+	// grab policy locks as we have ensured it doesn't already exist, so there
+	// will be no races as nothing else has this pointer. If 'force' was not used,
+	// an error would have been returned by now if the policy already existed
+	if pRaw != nil {
+		p = pRaw.(*Policy)
 	}
 	if p != nil {
-		return fmt.Errorf(fmt.Sprintf("key %q already exists", name))
+		p.l.Lock()
+		defer func() {
```
Comment (kalafut) [inline helper/keysutil/lock_manager.go]: Why is an anonymous function needed for this `defer`?

File: helper/keysutil/lock_manager.go [hunk 2]
```diff
@@ -103,26 +103,50 @@ func (lm *LockManager) RestorePolicy(ctx context.Context, storage logical.Storag
...
-	// cache will also be protected by the mutex above, so we don't need to
-	// re-check the cache later.
-	_, ok := lm.cache.Load(name)
-	if ok {
+	// If the policy is in cache and 'force' is not specified, error out. Anywhere
+	// that would put it in the cache will also be protected by the mutex above,
+	// so we don't need to re-check the cache later.
+	pRaw, ok := lm.cache.Load(name)
+	if ok && !force {
 		return fmt.Errorf(fmt.Sprintf("key %q already exists", name))
```
Comment (kalafut) [inline helper/keysutil/lock_manager.go]: Simpler: `fmt.Errorf("key %q already exists", name)`

Same on line 132.

**Comment by catsby** [comment]
Thanks for the review @kalafut , I made the suggested updates in d9550c5e4, please take another pass when you have a minute

File: helper/keysutil/lock_manager.go (hunk 1 above)
Comment (catsby) [reply helper/keysutil/lock_manager.go]: Probably bad copy-paste 😦

**Comment by kalafut** [comment]
LGTM!

**Pull request merged**